src/main/java/seedu/address/logic/commands/NoteCommand.java

```diff
@@ -0,0 +1,101 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NOTE;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.util.List;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Note;
+import seedu.address.model.person.Person;
+
+/**
+ * Adds a note of an existing person in the address book.
+ * A non-empty note must be specified.
+ */
+public class NoteCommand extends Command {
+
+    public static final String MESSAGE_ADD_NOTE_SUCCESS = "Added note to Person: %1$s";
+    public static final String COMMAND_WORD = "/note";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Adds note to person.\n"
+            + "Parameters: "
+            + PREFIX_NAME + "NAME, "
+            + PREFIX_NOTE + "NOTE "
+            + "Example: " + COMMAND_WORD + " Moochie" + " Meet at 6pm tuesday";
+    private final Name name;
+    private final Note note;
+
+    /**
+     * @param name of the person in the filtered person list to edit the note
+     * @param note of the person to be updated to
+     */
+    public NoteCommand(Name name, Note note) {
+        requireAllNonNull(name, note);
+        this.name = name;
+        this.note = note;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        Person personToEdit = findByName(lastShownList, name);
+        Person editedPerson = new Person(
```

Comment:
what about staff / supplier / maintainer edit? their special components such as skills would not get copied over.

Reply:
as part of v1.2 as it is an mvp, i have clarified the specifications such that note command can only be performed on the general person class. Otherwise, if note command is executed on any other class, it will throw an error. In the next iteration I will keep this in mind and expand on the note command.

Reply:
We can handle in future iteration using the method we proposed in discussion on handling "edit different type of Person", similar method can be used here.